### ISSUE TITLE
Crash with SIGSEGV in replaces_installed_package ()

### ISSUE THREAD

**Issue by sidt4** [openSUSE/libsolv#574]
It's not exactly clear what caused the crash, but PackageKit crashed when trying to download FC41 in GNOME Software in FC40.

On libsolv-0.7.30-1.fc40.x86_64:

```
(gdb) thread apply all bt

Thread 5 (Thread 0x7f922ae006c0 (LWP 1411)):
#0  0x00007f92399f087d in poll () at /lib64/libc.so.6
#1  0x00007f923a064c34 in g_main_context_iterate_unlocked.isra () at /lib64/libglib-2.0.so.0
#2  0x00007f923a004383 in g_main_context_iteration () at /lib64/libglib-2.0.so.0
#3  0x00007f923a0043e1 in glib_worker_main () at /lib64/libglib-2.0.so.0
#4  0x00007f923a033813 in g_thread_proxy () at /lib64/libglib-2.0.so.0
#5  0x00007f923997a6d7 in start_thread () at /lib64/libc.so.6
#6  0x00007f92399fe60c in clone3 () at /lib64/libc.so.6

Thread 4 (Thread 0x7f923954f000 (LWP 1399)):
#0  0x00007f92399f087d in poll () at /lib64/libc.so.6
#1  0x00007f923a064c34 in g_main_context_iterate_unlocked.isra () at /lib64/libglib-2.0.so.0
#2  0x00007f923a008f37 in g_main_loop_run () at /lib64/libglib-2.0.so.0
#3  0x0000556d8556cd56 in main (argc=<optimized out>, argv=<optimized out>) at ../src/pk-main.c:265

Thread 3 (Thread 0x7f922a4006c0 (LWP 1414)):
#0  0x00007f92399f087d in poll () at /lib64/libc.so.6
#1  0x00007f923a064c34 in g_main_context_iterate_unlocked.isra () at /lib64/libglib-2.0.so.0
#2  0x00007f923a008f37 in g_main_loop_run () at /lib64/libglib-2.0.so.0
#3  0x00007f9239e8d682 in gdbus_shared_thread_func.lto_priv () at /lib64/libgio-2.0.so.0
#4  0x00007f923a033813 in g_thread_proxy () at /lib64/libglib-2.0.so.0
#5  0x00007f923997a6d7 in start_thread () at /lib64/libc.so.6
#6  0x00007f92399fe60c in clone3 () at /lib64/libc.so.6

Thread 2 (Thread 0x7f922b8006c0 (LWP 1410)):
#0  0x00007f92399fc3dd in syscall () at /lib64/libc.so.6
#1  0x00007f923a061ccd in g_cond_wait () at /lib64/libglib-2.0.so.0
#2  0x00007f9239fcd61b in g_async_queue_pop_intern_unlocked () at /lib64/libglib-2.0.so.0
#3  0x00007f923a034a03 in g_thread_pool_spawn_thread () at /lib64/libglib-2.0.so.0
#4  0x00007f923a033813 in g_thread_proxy () at /lib64/libglib-2.0.so.0
#5  0x00007f923997a6d7 in start_thread () at /lib64/libc.so.6
#6  0x00007f92399fe60c in clone3 () at /lib64/libc.so.6

Thread 1 (Thread 0x7f9214c006c0 (LWP 4037)):
#0  0x00007f9229b6e33a in replaces_installed_package (pool=pool@entry=0x7f91d479dec0, p=p@entry=86615, noupdate=noupdate@entry=0x7f91d7ff8850) at /usr/src/debug/libsolv-0.7.30-1.fc40.x86_64/src/solver.c:1162
#1  0x00007f9229b74461 in solver_run_sat (solv=solv@entry=0x7f91d7ff8750, disablerules=disablerules@entry=1, doweak=1) at /usr/src/debug/libsolv-0.7.30-1.fc40.x86_64/src/solver.c:3094
#2  0x00007f9229b77ea8 in solver_solve (solv=0x7f91d7ff8750, job=0x7f91d7ff8758) at /usr/src/debug/libsolv-0.7.30-1.fc40.x86_64/src/solver.c:4440
#3  0x00007f922af4a730 in libdnf::Goal::Impl::solve (this=0x7f91ec013f60, job=0x7f91dcb6f0f0, flags=<optimized out>) at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/goal/Goal.cpp:1579
#4  0x00007f922af4a812 in libdnf::Goal::run (this=0x7f91ec00ba60, flags=DNF_ALLOW_UNINSTALL) at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/goal/IdQueue.hpp:68
#5  0x00007f922aef1e4d in hy_goal_run_flags (goal=<optimized out>, flags=<optimized out>) at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/hy-goal.cpp:248
#6  0x00007f922af12b62 in dnf_goal_depsolve (goal=0x7f91ec00ba60, flags=DNF_ALLOW_UNINSTALL, error=error@entry=0x7f9214bffc88) at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/dnf-goal.cpp:105
#7  0x00007f923953e615 in pk_backend_transaction_run (job=job@entry=0x556db6084100, state=0x7f91ec0d0030, error=error@entry=0x7f9214bffc88) at ../backends/dnf/pk-backend-dnf.c:2579
#8  0x00007f92395445a5 in pk_backend_upgrade_system_thread (job=0x556db6084100, params=<optimized out>, user_data=<optimized out>) at ../backends/dnf/pk-backend-dnf.c:3496
#9  0x0000556d85584f26 in pk_backend_job_thread_setup (thread_data=0x556db60f90b0) at ../src/pk-backend-job.c:743
#10 0x00007f923a033813 in g_thread_proxy () at /lib64/libglib-2.0.so.0
#11 0x00007f923997a6d7 in start_thread () at /lib64/libc.so.6
#12 0x00007f92399fe60c in clone3 () at /lib64/libc.so.6
(gdb) 
```

```
(gdb) bt full
#0  0x00007f9229b6e33a in replaces_installed_package (pool=pool@entry=0x7f91d479dec0, p=p@entry=86615, noupdate=noupdate@entry=0x7f91d7ff8850)
    at /usr/src/debug/libsolv-0.7.30-1.fc40.x86_64/src/solver.c:1162
        installed = 0x7f91ec1a8480
        s = 0x7f91cda4d298
        s2 = 0x7f91cd5c56e8
        p2 = <optimized out>
        pp2 = <optimized out>
        obs = -2147108125
        obsp = 0x7f91de51b264
#1  0x00007f9229b74461 in solver_run_sat (solv=solv@entry=0x7f91d7ff8750, disablerules=disablerules@entry=1, doweak=1)
    at /usr/src/debug/libsolv-0.7.30-1.fc40.x86_64/src/solver.c:3094
        l = 187
        lastsi = <optimized out>
        starti = 0
        endi = <optimized out>
        lasti = <optimized out>
        lastiend = <optimized out>
        dq = {
          elements = 0x7f91ef1ad6e0,
          count = 0,
          alloc = 0x7f91ef1ad6e0,
          left = 86
        }
        dqs = {
          elements = 0x7f91cc161a40,
          count = 0,
          alloc = 0x7f91cc161a40,
          left = 4
        }
        systemlevel = <optimized out>
        level = <optimized out>
        olevel = <optimized out>
        r = <optimized out>
        i = <optimized out>
        s = <optimized out>
        pool = 0x7f91d479dec0
        p = 86615
        minimizationsteps = 0
        __PRETTY_FUNCTION__ = "solver_run_sat"
#2  0x00007f9229b77ea8 in solver_solve (solv=0x7f91d7ff8750, job=0x7f91d7ff8758) at /usr/src/debug/libsolv-0.7.30-1.fc40.x86_64/src/solver.c:4440
        pool = 0x7f91d479dec0
        installed = 0x7f91ec1a8480
        i = <optimized out>
        oldnrules = <optimized out>
        initialnrules = <optimized out>
        addedmap = {
          map = 0x0,
          size = 0
        }
        installcandidatemap = {
          map = 0x0,
          size = 0
        }
        how = <optimized out>
        what = <optimized out>
        select = <optimized out>
        name = <optimized out>
        weak = <optimized out>
        p = <optimized out>
        pp = <optimized out>
        d = <optimized out>
        q = {
          elements = 0x0,
          count = 0,
          alloc = 0x0,
          left = 0
        }
        s = <optimized out>
        name_s = <optimized out>
        r = <optimized out>
        now = 465516413
        solve_start = 465516339
        needduprules = <optimized out>
        hasbestinstalljob = <optimized out>
        hasfavorjob = <optimized out>
        haslockjob = <optimized out>
        hasblacklistjob = <optimized out>
        hasexcludefromweakjob = <optimized out>
        __PRETTY_FUNCTION__ = "solver_solve"
#3  0x00007f922af4a730 in libdnf::Goal::Impl::solve (this=0x7f91ec013f60, job=0x7f91dcb6f0f0, flags=<optimized out>)
    at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/goal/Goal.cpp:1579
        solv = 0x7f91d7ff8750
#4  0x00007f922af4a812 in libdnf::Goal::run (this=0x7f91ec00ba60, flags=DNF_ALLOW_UNINSTALL)
    at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/goal/IdQueue.hpp:68
        job = std::unique_ptr<libdnf::IdQueue> = {
          get() = 0x7f91dcb6f0f0
        }
        ret = <optimized out>
#5  0x00007f922aef1e4d in hy_goal_run_flags (goal=<optimized out>, flags=<optimized out>)
    at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/hy-goal.cpp:248
#6  0x00007f922af12b62 in dnf_goal_depsolve (goal=0x7f91ec00ba60, flags=DNF_ALLOW_UNINSTALL, error=error@entry=0x7f9214bffc88)
    at /usr/src/debug/libdnf-0.73.3-1.fc40.x86_64/libdnf/dnf-goal.cpp:105
        j = <optimized out>
        string = 0x0
        moduleContainer = <optimized out>
        query = {
          pImpl = std::unique_ptr<class libdnf::Query::Impl> = {
            get() = 0x7f91ec174850
          }
        }
        pkgset = {
          pImpl = std::unique_ptr<class libdnf::PackageSet::Impl> = {
            get() = 0x7f91cc19a620
          }
        }
        cnt = <optimized out>
        rc = <optimized out>
        sack = 0x7f91ec1a8e90
        protected_packages = <optimized out>
        cprotected_packages = std::vector of length 14, capacity 14 = {0x556db5fea4d0 "dnf", 0x556db5fea4f0 "NetworkManager", 0x556db5fea510 "gnome-shell", 0x556db5fea530 "systemd", 0x556db5fea550 "systemd-udev", 0x556db5f9e120 "grub2-tools-minimal", 0x556db5fea590 "grub2-efi-x64", 0x556db5fea5b0 "sudo", 0x556db5fea5d0 "shim-aa64", 0x556db5fea5f0 "shim-arm", 0x556db5fea610 "shim-ia32", 0x556db5fea630 "shim-x64", 0x556db5fea650 "setup", 0x0}
#7  0x00007f923953e615 in pk_backend_transaction_run (job=job@entry=0x556db6084100, state=0x7f91ec0d0030, error=error@entry=0x7f9214bffc88)
    at ../backends/dnf/pk-backend-dnf.c:2579
        state_local = 0x7f91ec282250
        dnf_flags = <optimized out>
        job_data = 0x556db60f8fb0
        ret = <optimized out>
        flags = <optimized out>
#8  0x00007f92395445a5 in pk_backend_upgrade_system_thread (job=0x556db6084100, params=<optimized out>, user_data=<optimized out>)
    at ../backends/dnf/pk-backend-dnf.c:3496
        state_local = <optimized out>
        job_data = 0x556db60f8fb0
        priv = <optimized out>
        filters = 8
        ret = <optimized out>
        release_ver = 0x556db60b8478 "41"
        sack = 0x7f91ec1a8e90
        error = 0x0
        __func__ = "pk_backend_upgrade_system_thread"
#9  0x0000556d85584f26 in pk_backend_job_thread_setup (thread_data=0x556db60f90b0) at ../src/pk-backend-job.c:743
        helper = 0x556db60f90b0
#10 0x00007f923a033813 in g_thread_proxy () at /lib64/libglib-2.0.so.0
#11 0x00007f923997a6d7 in start_thread () at /lib64/libc.so.6
#12 0x00007f92399fe60c in clone3 () at /lib64/libc.so.
```

```
(gdb) disassemble 
Dump of assembler code for function replaces_installed_package:
   0x00007f9229b6e130 <+0>:	endbr64
   0x00007f9229b6e134 <+4>:	push   %rbp
   0x00007f9229b6e135 <+5>:	mov    %rsp,%rbp
   0x00007f9229b6e138 <+8>:	push   %r15
   0x00007f9229b6e13a <+10>:	push   %r14
   0x00007f9229b6e13c <+12>:	mov    %esi,%r14d
   0x00007f9229b6e13f <+15>:	push   %r13
   0x00007f9229b6e141 <+17>:	mov    %rdx,%r13
   0x00007f9229b6e144 <+20>:	movslq %esi,%rdx
   0x00007f9229b6e147 <+23>:	push   %r12
   0x00007f9229b6e149 <+25>:	lea    0x0(,%rdx,8),%rax
   0x00007f9229b6e151 <+33>:	push   %rbx
   0x00007f9229b6e152 <+34>:	sub    %rdx,%rax
   0x00007f9229b6e155 <+37>:	sub    $0x28,%rsp
   0x00007f9229b6e159 <+41>:	mov    0x60(%rdi),%rdx
   0x00007f9229b6e15d <+45>:	mov    0x58(%rdi),%rbx
   0x00007f9229b6e161 <+49>:	lea    (%rdx,%rax,8),%r12
   0x00007f9229b6e165 <+53>:	cmp    %rbx,0x10(%r12)
   0x00007f9229b6e16a <+58>:	je     0x7f9229b6e420 <replaces_installed_package+752>
   0x00007f9229b6e170 <+64>:	mov    (%r12),%esi
   0x00007f9229b6e174 <+68>:	test   %esi,%esi
   0x00007f9229b6e176 <+70>:	js     0x7f9229b6e240 <replaces_installed_package+272>
   0x00007f9229b6e17c <+76>:	mov    0xc0(%rdi),%rdx
   0x00007f9229b6e183 <+83>:	movslq %esi,%rax
   0x00007f9229b6e186 <+86>:	mov    (%rdx,%rax,4),%eax
   0x00007f9229b6e189 <+89>:	test   %eax,%eax
   0x00007f9229b6e18b <+91>:	jne    0x7f9229b6e19a <replaces_installed_package+106>
   0x00007f9229b6e18d <+93>:	mov    %rdi,-0x38(%rbp)
   0x00007f9229b6e191 <+97>:	call   0x7f9229b66d50 <pool_addrelproviders@plt>
   0x00007f9229b6e196 <+102>:	mov    -0x38(%rbp),%rdi
   0x00007f9229b6e19a <+106>:	mov    0xd0(%rdi),%r8
   0x00007f9229b6e1a1 <+113>:	cltq
   0x00007f9229b6e1a3 <+115>:	lea    0x0(,%rax,4),%rdx
   0x00007f9229b6e1ab <+123>:	movslq (%r8,%rax,4),%rax
   0x00007f9229b6e1af <+127>:	test   %eax,%eax
   0x00007f9229b6e1b1 <+129>:	je     0x7f9229b6e260 <replaces_installed_package+304>
   0x00007f9229b6e1b7 <+135>:	mov    0x60(%rdi),%rsi
   0x00007f9229b6e1bb <+139>:	lea    0x4(%r8,%rdx,1),%rcx
   0x00007f9229b6e1c0 <+144>:	jmp    0x7f9229b6e1eb <replaces_installed_package+187>
   0x00007f9229b6e1c2 <+146>:	data16 cs nopw 0x0(%rax,%rax,1)
   0x00007f9229b6e1cd <+157>:	data16 cs nopw 0x0(%rax,%rax,1)
   0x00007f9229b6e1d8 <+168>:	nopl   0x0(%rax,%rax,1)
   0x00007f9229b6e1e0 <+176>:	movslq (%rcx),%rax
   0x00007f9229b6e1e3 <+179>:	add    $0x4,%rcx
   0x00007f9229b6e1e7 <+183>:	test   %eax,%eax
   0x00007f9229b6e1e9 <+185>:	je     0x7f9229b6e260 <replaces_installed_package+304>
   0x00007f9229b6e1eb <+187>:	lea    0x0(,%rax,8),%rdx
   0x00007f9229b6e1f3 <+195>:	sub    %rax,%rdx
   0x00007f9229b6e1f6 <+198>:	lea    (%rsi,%rdx,8),%rax
   0x00007f9229b6e1fa <+202>:	cmp    %rbx,0x10(%rax)
   0x00007f9229b6e1fe <+206>:	jne    0x7f9229b6e1e0 <replaces_installed_package+176>
   0x00007f9229b6e200 <+208>:	mov    (%r12),%edx
   0x00007f9229b6e204 <+212>:	cmp    %edx,(%rax)
   0x00007f9229b6e206 <+214>:	jne    0x7f9229b6e1e0 <replaces_installed_package+176>
   0x00007f9229b6e208 <+216>:	test   %r13,%r13
   0x00007f9229b6e20b <+219>:	je     0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e20d <+221>:	mov    %r14d,%eax
   0x00007f9229b6e210 <+224>:	sub    0x20(%rbx),%eax
   0x00007f9229b6e213 <+227>:	mov    0x0(%r13),%r9
   0x00007f9229b6e217 <+231>:	mov    %eax,%edx
   0x00007f9229b6e219 <+233>:	and    $0x7,%eax
   0x00007f9229b6e21c <+236>:	sar    $0x3,%edx
   0x00007f9229b6e21f <+239>:	movslq %edx,%rdx
   0x00007f9229b6e222 <+242>:	movzbl (%r9,%rdx,1),%edx
   0x00007f9229b6e227 <+247>:	bt     %eax,%edx
   0x00007f9229b6e22a <+250>:	jb     0x7f9229b6e1e0 <replaces_installed_package+176>
   0x00007f9229b6e22c <+252>:	add    $0x28,%rsp
   0x00007f9229b6e230 <+256>:	mov    $0x1,%eax
   0x00007f9229b6e235 <+261>:	pop    %rbx
   0x00007f9229b6e236 <+262>:	pop    %r12
   0x00007f9229b6e238 <+264>:	pop    %r13
   0x00007f9229b6e23a <+266>:	pop    %r14
   0x00007f9229b6e23c <+268>:	pop    %r15
   0x00007f9229b6e23e <+270>:	pop    %rbp
   0x00007f9229b6e23f <+271>:	ret
   0x00007f9229b6e240 <+272>:	mov    0xc8(%rdi),%rdx
   0x00007f9229b6e247 <+279>:	lea    -0x80000000(%rsi),%eax
   0x00007f9229b6e24d <+285>:	cltq
   0x00007f9229b6e24f <+287>:	mov    (%rdx,%rax,4),%eax
   0x00007f9229b6e252 <+290>:	test   %eax,%eax
   0x00007f9229b6e254 <+292>:	je     0x7f9229b6e18d <replaces_installed_package+93>
   0x00007f9229b6e25a <+298>:	jmp    0x7f9229b6e19a <replaces_installed_package+106>
   0x00007f9229b6e25f <+303>:	nop
   0x00007f9229b6e260 <+304>:	mov    0x1c(%r12),%eax
   0x00007f9229b6e265 <+309>:	test   %eax,%eax
   0x00007f9229b6e267 <+311>:	je     0x7f9229b6e309 <replaces_installed_package+473>
   0x00007f9229b6e26d <+317>:	mov    0x10(%r12),%rdx
   0x00007f9229b6e272 <+322>:	mov    0x38(%rdx),%rdx
   0x00007f9229b6e276 <+326>:	lea    (%rdx,%rax,4),%rax
   0x00007f9229b6e27a <+330>:	mov    (%rax),%r15d
   0x00007f9229b6e27d <+333>:	lea    0x4(%rax),%rsi
   0x00007f9229b6e281 <+337>:	mov    %rsi,-0x38(%rbp)
   0x00007f9229b6e285 <+341>:	test   %r15d,%r15d
   0x00007f9229b6e288 <+344>:	je     0x7f9229b6e309 <replaces_installed_package+473>
   0x00007f9229b6e28a <+346>:	test   %r15d,%r15d
   0x00007f9229b6e28d <+349>:	js     0x7f9229b6e468 <replaces_installed_package+824>
   0x00007f9229b6e293 <+355>:	mov    0xc0(%rdi),%rcx
   0x00007f9229b6e29a <+362>:	movslq %r15d,%rax
   0x00007f9229b6e29d <+365>:	mov    (%rcx,%rax,4),%eax
   0x00007f9229b6e2a0 <+368>:	test   %eax,%eax
   0x00007f9229b6e2a2 <+370>:	jne    0x7f9229b6e2bb <replaces_installed_package+395>
   0x00007f9229b6e2a4 <+372>:	mov    %r15d,%esi
   0x00007f9229b6e2a7 <+375>:	mov    %rdi,-0x40(%rbp)
   0x00007f9229b6e2ab <+379>:	call   0x7f9229b66d50 <pool_addrelproviders@plt>
   0x00007f9229b6e2b0 <+384>:	mov    -0x40(%rbp),%rdi
   0x00007f9229b6e2b4 <+388>:	mov    0xd0(%rdi),%r8
   0x00007f9229b6e2bb <+395>:	movslq %eax,%rcx
   0x00007f9229b6e2be <+398>:	movslq (%r8,%rcx,4),%rax
   0x00007f9229b6e2c2 <+402>:	lea    0x4(,%rcx,4),%rcx
   0x00007f9229b6e2ca <+410>:	test   %eax,%eax
   0x00007f9229b6e2cc <+412>:	je     0x7f9229b6e2f7 <replaces_installed_package+455>
   0x00007f9229b6e2ce <+414>:	lea    0x0(,%rax,8),%rsi
   0x00007f9229b6e2d6 <+422>:	sub    %rax,%rsi
   0x00007f9229b6e2d9 <+425>:	mov    0x60(%rdi),%rax
   0x00007f9229b6e2dd <+429>:	lea    (%rax,%rsi,8),%r9
   0x00007f9229b6e2e1 <+433>:	mov    0x58(%rdi),%rax
   0x00007f9229b6e2e5 <+437>:	cmp    %rax,0x10(%r9)
   0x00007f9229b6e2e9 <+441>:	je     0x7f9229b6e320 <replaces_installed_package+496>
   0x00007f9229b6e2eb <+443>:	movslq (%r8,%rcx,1),%rax
   0x00007f9229b6e2ef <+447>:	add    $0x4,%rcx
   0x00007f9229b6e2f3 <+451>:	test   %eax,%eax
   0x00007f9229b6e2f5 <+453>:	jne    0x7f9229b6e2ce <replaces_installed_package+414>
   0x00007f9229b6e2f7 <+455>:	addq   $0x4,-0x38(%rbp)
   0x00007f9229b6e2fc <+460>:	mov    -0x38(%rbp),%rax
   0x00007f9229b6e300 <+464>:	mov    -0x4(%rax),%r15d
   0x00007f9229b6e304 <+468>:	test   %r15d,%r15d
   0x00007f9229b6e307 <+471>:	jne    0x7f9229b6e28a <replaces_installed_package+346>
   0x00007f9229b6e309 <+473>:	add    $0x28,%rsp
   0x00007f9229b6e30d <+477>:	xor    %eax,%eax
   0x00007f9229b6e30f <+479>:	pop    %rbx
   0x00007f9229b6e310 <+480>:	pop    %r12
   0x00007f9229b6e312 <+482>:	pop    %r13
   0x00007f9229b6e314 <+484>:	pop    %r14
   0x00007f9229b6e316 <+486>:	pop    %r15
   0x00007f9229b6e318 <+488>:	pop    %rbp
   0x00007f9229b6e319 <+489>:	ret
   0x00007f9229b6e31a <+490>:	nopw   0x0(%rax,%rax,1)
   0x00007f9229b6e320 <+496>:	test   %r13,%r13
   0x00007f9229b6e323 <+499>:	je     0x7f9229b6e344 <replaces_installed_package+532>
   0x00007f9229b6e325 <+501>:	mov    %r14d,%eax
   0x00007f9229b6e328 <+504>:	sub    0x20(%rbx),%eax
   0x00007f9229b6e32b <+507>:	mov    0x0(%r13),%r10
   0x00007f9229b6e32f <+511>:	mov    %eax,%esi
   0x00007f9229b6e331 <+513>:	and    $0x7,%eax
   0x00007f9229b6e334 <+516>:	sar    $0x3,%esi
   0x00007f9229b6e337 <+519>:	movslq %esi,%rsi
=> 0x00007f9229b6e33a <+522>:	movzbl (%r10,%rsi,1),%esi
   0x00007f9229b6e33f <+527>:	bt     %eax,%esi
   0x00007f9229b6e342 <+530>:	jb     0x7f9229b6e2eb <replaces_installed_package+443>
   0x00007f9229b6e344 <+532>:	mov    0x160(%rdi),%edx
   0x00007f9229b6e34a <+538>:	test   %edx,%edx
   0x00007f9229b6e34c <+540>:	je     0x7f9229b6e450 <replaces_installed_package+800>
   0x00007f9229b6e352 <+546>:	mov    0x168(%rdi),%eax
   0x00007f9229b6e358 <+552>:	test   %eax,%eax
   0x00007f9229b6e35a <+554>:	je     0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e360 <+560>:	mov    0x4(%r12),%eax
   0x00007f9229b6e365 <+565>:	mov    0x4(%r9),%esi
   0x00007f9229b6e369 <+569>:	cmp    %esi,%eax
   0x00007f9229b6e36b <+571>:	je     0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e371 <+577>:	mov    0x90(%rdi),%r11d
   0x00007f9229b6e378 <+584>:	cmp    %r11d,%eax
   0x00007f9229b6e37b <+587>:	jae    0x7f9229b6e4ce <replaces_installed_package+926>
   0x00007f9229b6e381 <+593>:	mov    0x88(%rdi),%r10
   0x00007f9229b6e388 <+600>:	test   %r10,%r10
   0x00007f9229b6e38b <+603>:	je     0x7f9229b6e39e <replaces_installed_package+622>
   0x00007f9229b6e38d <+605>:	movslq %eax,%r8
   0x00007f9229b6e390 <+608>:	movzbl (%r10,%r8,1),%r8d
   0x00007f9229b6e395 <+613>:	test   %r8b,%r8b
   0x00007f9229b6e398 <+616>:	jne    0x7f9229b6e490 <replaces_installed_package+864>
   0x00007f9229b6e39e <+622>:	mov    %eax,%esi
   0x00007f9229b6e3a0 <+624>:	mov    %rcx,-0x50(%rbp)
   0x00007f9229b6e3a4 <+628>:	mov    %r9,-0x48(%rbp)
   0x00007f9229b6e3a8 <+632>:	mov    %rdi,-0x40(%rbp)
   0x00007f9229b6e3ac <+636>:	call   0x7f9229b675e0 <pool_arch2color_slow@plt>
   0x00007f9229b6e3b1 <+641>:	mov    -0x48(%rbp),%r9
   0x00007f9229b6e3b5 <+645>:	mov    -0x40(%rbp),%rdi
   0x00007f9229b6e3b9 <+649>:	mov    -0x50(%rbp),%rcx
   0x00007f9229b6e3bd <+653>:	mov    %eax,%r8d
   0x00007f9229b6e3c0 <+656>:	mov    0x4(%r9),%esi
   0x00007f9229b6e3c4 <+660>:	cmp    0x90(%rdi),%esi
   0x00007f9229b6e3ca <+666>:	jae    0x7f9229b6e409 <replaces_installed_package+729>
   0x00007f9229b6e3cc <+668>:	mov    0x88(%rdi),%r10
   0x00007f9229b6e3d3 <+675>:	test   %r10,%r10
   0x00007f9229b6e3d6 <+678>:	je     0x7f9229b6e3e8 <replaces_installed_package+696>
   0x00007f9229b6e3d8 <+680>:	movslq %esi,%rax
   0x00007f9229b6e3db <+683>:	movzbl (%r10,%rax,1),%eax
   0x00007f9229b6e3e0 <+688>:	test   %al,%al
   0x00007f9229b6e3e2 <+690>:	jne    0x7f9229b6e488 <replaces_installed_package+856>
   0x00007f9229b6e3e8 <+696>:	mov    %rcx,-0x50(%rbp)
   0x00007f9229b6e3ec <+700>:	mov    %r8b,-0x48(%rbp)
   0x00007f9229b6e3f0 <+704>:	mov    %rdi,-0x40(%rbp)
   0x00007f9229b6e3f4 <+708>:	call   0x7f9229b675e0 <pool_arch2color_slow@plt>
   0x00007f9229b6e3f9 <+713>:	movzbl -0x48(%rbp),%r8d
   0x00007f9229b6e3fe <+718>:	mov    -0x50(%rbp),%rcx
   0x00007f9229b6e402 <+722>:	mov    -0x40(%rbp),%rdi
   0x00007f9229b6e406 <+726>:	and    %eax,%r8d
   0x00007f9229b6e409 <+729>:	test   %r8b,%r8b
   0x00007f9229b6e40c <+732>:	jne    0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e412 <+738>:	mov    0xd0(%rdi),%r8
   0x00007f9229b6e419 <+745>:	jmp    0x7f9229b6e2eb <replaces_installed_package+443>
   0x00007f9229b6e41e <+750>:	xchg   %ax,%ax
   0x00007f9229b6e420 <+752>:	test   %r13,%r13
   0x00007f9229b6e423 <+755>:	je     0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e429 <+761>:	mov    %esi,%eax
   0x00007f9229b6e42b <+763>:	sub    0x20(%rbx),%eax
   0x00007f9229b6e42e <+766>:	mov    0x0(%r13),%rcx
   0x00007f9229b6e432 <+770>:	mov    %eax,%edx
   0x00007f9229b6e434 <+772>:	and    $0x7,%eax
   0x00007f9229b6e437 <+775>:	sar    $0x3,%edx
   0x00007f9229b6e43a <+778>:	movslq %edx,%rdx
   0x00007f9229b6e43d <+781>:	movzbl (%rcx,%rdx,1),%edx
   0x00007f9229b6e441 <+785>:	bt     %eax,%edx
   0x00007f9229b6e444 <+788>:	jb     0x7f9229b6e170 <replaces_installed_package+64>
   0x00007f9229b6e44a <+794>:	jmp    0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e44f <+799>:	nop
   0x00007f9229b6e450 <+800>:	test   %r15d,%r15d
   0x00007f9229b6e453 <+803>:	js     0x7f9229b6e49e <replaces_installed_package+878>
   0x00007f9229b6e455 <+805>:	cmp    %r15d,(%r9)
   0x00007f9229b6e458 <+808>:	je     0x7f9229b6e352 <replaces_installed_package+546>
   0x00007f9229b6e45e <+814>:	jmp    0x7f9229b6e2eb <replaces_installed_package+443>
   0x00007f9229b6e463 <+819>:	nopl   0x0(%rax,%rax,1)
   0x00007f9229b6e468 <+824>:	mov    0xc8(%rdi),%rcx
   0x00007f9229b6e46f <+831>:	lea    -0x80000000(%r15),%eax
   0x00007f9229b6e476 <+838>:	cltq
   0x00007f9229b6e478 <+840>:	mov    (%rcx,%rax,4),%eax
   0x00007f9229b6e47b <+843>:	test   %eax,%eax
   0x00007f9229b6e47d <+845>:	je     0x7f9229b6e2a4 <replaces_installed_package+372>
   0x00007f9229b6e483 <+851>:	jmp    0x7f9229b6e2bb <replaces_installed_package+395>
   0x00007f9229b6e488 <+856>:	and    %eax,%r8d
   0x00007f9229b6e48b <+859>:	jmp    0x7f9229b6e409 <replaces_installed_package+729>
   0x00007f9229b6e490 <+864>:	cmp    %r11d,%esi
   0x00007f9229b6e493 <+867>:	jb     0x7f9229b6e3d8 <replaces_installed_package+680>
   0x00007f9229b6e499 <+873>:	jmp    0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e49e <+878>:	mov    %r9,%rsi
   0x00007f9229b6e4a1 <+881>:	mov    %r15d,%edx
   0x00007f9229b6e4a4 <+884>:	mov    %rcx,-0x50(%rbp)
   0x00007f9229b6e4a8 <+888>:	mov    %r9,-0x48(%rbp)
   0x00007f9229b6e4ac <+892>:	mov    %rdi,-0x40(%rbp)
   0x00007f9229b6e4b0 <+896>:	call   0x7f9229b673d0 <pool_match_nevr_rel@plt>
   0x00007f9229b6e4b5 <+901>:	mov    -0x40(%rbp),%rdi
   0x00007f9229b6e4b9 <+905>:	mov    -0x48(%rbp),%r9
   0x00007f9229b6e4bd <+909>:	test   %eax,%eax
   0x00007f9229b6e4bf <+911>:	mov    -0x50(%rbp),%rcx
   0x00007f9229b6e4c3 <+915>:	jne    0x7f9229b6e352 <replaces_installed_package+546>
   0x00007f9229b6e4c9 <+921>:	jmp    0x7f9229b6e412 <replaces_installed_package+738>
   0x00007f9229b6e4ce <+926>:	cmp    %r11d,%esi
   0x00007f9229b6e4d1 <+929>:	jae    0x7f9229b6e22c <replaces_installed_package+252>
   0x00007f9229b6e4d7 <+935>:	mov    $0xffffffff,%r8d
   0x00007f9229b6e4dd <+941>:	jmp    0x7f9229b6e3cc <replaces_installed_package+668>
End of assembler dump.
(gdb) 
```

**Comment by mlschroe** [comment]
I think I know what's going on. Will fix ASAP.

**Comment by mlschroe** [comment]
Fixed in libsolv-0.7.31
